core/src/main/resources/org/apache/spark/ui/static/webui.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+function collapseTablePageLoad(name, table){
```

Comment:
I didn't look into any standard frameworks, so there might be some.

I used a modified version of the JS the Web UI uses for showing/hiding the Additional Metrics, DAG Viz, and Timeline. Since each of those used slightly different and hard-coded versions of the same code for each I wanted to make a more reusable version here. 

If we want I can look into finding a preexisting library to do this instead.